internal/metrics/v2/gauge.go

```diff
@@ -0,0 +1,65 @@
+package v2
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// Gauge is a metrics gauge.
+// Please consult Prometheus documentation for a detailed understanding.
+type Gauge interface {
+	// Set sets the Gauge to an arbitrary value.
+	Set(float64, ...Label)
+	// Inc increments the Gauge by 1. Use Add to increment it by arbitrary
+	// values.
+	Inc(...Label)
+	// Dec decrements the Gauge by 1. Use Sub to decrement it by arbitrary
+	// values.
+	Dec(...Label)
+	// Add adds the given value to the Gauge. (The value can be negative,
+	// resulting in a decrease of the Gauge.)
+	Add(float64, ...Label)
+	// Sub subtracts the given value from the Gauge. (The value can be
+	// negative, resulting in an increase of the Gauge.)
+	Sub(float64, ...Label)
+}
+
+type GaugeOpts Opts
+
+func NewGauge(registry prometheus.Registerer, opts GaugeOpts) Gauge {
+	return &prometheusGauge{
+		gauge: promauto.With(registry).
+			NewGaugeVec(prometheus.GaugeOpts{
+				Namespace: namespace,
+				Subsystem: opts.Subsystem,
+				Name:      opts.Name,
+				Help:      opts.Help,
+			},
+				opts.LabelNames,
+			),
+	}
```

Comment:
nit: The formatting here (and in the other places) is really bizarre:
```suggestion
	return &prometheusGauge{
		gauge: promauto.With(registry).NewGaugeVec(
			prometheus.GaugeOpts{
				Namespace: namespace,
				Subsystem: opts.Subsystem,
				Name:      opts.Name,
				Help:      opts.Help,
			},
			opts.LabelNames,
		),
	}
```

Reply:
interesting that it made it through gofmt, I'll double check

Reply:
ah looks like I already addressed that ;-)

Reply:
It's technically still valid, but yeah, it's not ideal. I don't see this being corrected, the indention is still very screwy & hard to tell the args being passed into `NewGaugeVec()` due to the formatting.